weather_mv/loader_pipeline/ee_util.py

```diff
@@ -0,0 +1,457 @@
+                               ' will be pushed.')
+        subparser.add_argument('-s', '--skip-region-validation', action='store_true', default=False,
+                               help='Skip validation of regions for data migration. Default: off')
+        subparser.add_argument('--tiff_location', type=str, default=None,
+                               help='The GCS location where the tiff files will be pushed.')
+        subparser.add_argument('-u', '--use_personal_account', action='store_true', default=False,
+                               help='To use personal account for earth engine authentication.')
+
+    @classmethod
+    def validate_arguments(cls, known_args: argparse.Namespace, pipeline_args: t.List[str]) -> None:
+        pipeline_options = PipelineOptions(pipeline_args)
+        pipeline_options_dict = pipeline_options.get_all_options()
+
+        if not known_args.tiff_location:
+            raise RuntimeError("'--tiff_location' is required for ee ingestion.")
+        if not known_args.ee_asset:
+            raise RuntimeError("'--ee_asset' is required for ee ingestion.")
+        if not re.match("^projects/.+/assets.*", known_args.ee_asset):
+            raise RuntimeError("'--ee_asset' is required to be in format: projects/+/assets/*.")
+
+        # Check that Cloud resource regions are consistent.
+        if not (known_args.dry_run or known_args.skip_region_validation):
+            # Program execution will terminate on failure of region validation.
+            logger.info('Validating regions for data migration. This might take a few seconds...')
+            validate_region(temp_location=pipeline_options_dict.get('temp_location'),
+                            region=pipeline_options_dict.get('region'))
+            logger.info('Region validation completed successfully.')
+
+    def __post_init__(self):
+        """Initializes Sink by initialize earthengine based on user input."""
+
+        if self.dry_run:
+            logger.debug('Uploaded assets into earth engine ...')
+            return
+
+    def expand(self, paths):
+        """Convert gribs from paths into tiff and upload them into earth engine."""
+        (
+            paths
+            | 'ExtractGribData' >> beam.ParDo(
+                ExtractGribData(
+                    open_dataset_kwargs=self.xarray_open_dataset_kwargs,
+                    disable_in_memory_copy=self.disable_in_memory_copy,
+                    disable_grib_schema_normalization=self.disable_grib_schema_normalization))
+            | 'IngestIntoEE' >> beam.ParDo(
+                IngestIntoEE(
+                    dry_run=self.dry_run,
+                    ee_asset=self.ee_asset,
+                    tiff_location=self.tiff_location,
+                    use_personal_account=self.use_personal_account))
+        )
+
+
+@dataclasses.dataclass
+class ExtractGribData(beam.DoFn):
```

Comment:
How about in this transform, we: 
- download the grib file
- convert it to a COG locally
- upload the tif to GCS
Then, the next transform can be focused on registering the asset with EE.

Reply:
My main concern: It's expensive to serialize and deserialize the grib data when we could perform all the IO in the same VM.

Reply:
Agreed. Updated it.